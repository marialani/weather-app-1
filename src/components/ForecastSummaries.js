import React from "react";
import PropTypes from "prop-types";
import ForecastSummary from "./ForecastSummary";
import "../styles/ForecastSummaries.css";

const ForecastSummaries = ({ forecasts, onForecastSelect }) => (
  <div className="forecast-summaries">
    {forecasts.map((forecast) => (
      <ForecastSummary
        key={forecast.date}
        date={forecast.date}
        description={forecast.description}
        icon={forecast.icon.toString()}
        temperature={forecast.temperature}
        onSelect={onForecastSelect}
      />
    ))}
  </div>
);

ForecastSummaries.propTypes = {
  onForecastSelect: PropTypes.func.isRequired,
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number.isRequired,
      description: PropTypes.string.isRequired,
      icon: PropTypes.number.isRequired,
      temperature: PropTypes.shape({
        min: PropTypes.number,
        max: PropTypes.number,
      }).isRequired,
    })
  ).isRequired,
};

export default ForecastSummaries;
