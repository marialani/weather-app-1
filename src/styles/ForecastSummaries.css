.forecast-summaries {
  display: flex;
  justify-content: space-between;
  margin: 4rem;
  font-size: 25px;
}
