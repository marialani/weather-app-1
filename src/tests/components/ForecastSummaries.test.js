import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("Forecast Summaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 1525046400000,
        description: "stub description",
        icon: 800,
        temperature: { min: 12, max: 18 },
      },
      {
        date: 1525132800000,
        description: "stub description sequel",
        icon: 211,
        temperature: { min: 14, max: 22 },
      },
    ],
    onForecastSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
