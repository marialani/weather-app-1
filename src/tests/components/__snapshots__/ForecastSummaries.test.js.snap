// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Summaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Mon 30th Apr
      </div>
      <div
        class="forecast-summary_temperature"
      >
        18°C
      </div>
      <div
        class="forecast-summary_description"
      >
        stub description
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Tue 1st May
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary_description"
      >
        stub description sequel
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-thunderstorm"
          name="owm"
        />
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
